### PR TITLE
create peer when pre-vote received if need

### PR DESCRIPTION
**Pull request by hicqu**
## What have you changed? (mandatory)
Create new peer if pre-vote received and there is no such peer. The bad case is showed in the new added test.

## What are the type of the changes? (mandatory)
Bug fix.

## How has this PR been tested? (mandatory)
make dev.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No.

### CODE REVIEW

**Comment by siddontang** [comment]
Great Catch

File: tests/raftstore_cases/mod.rs [hunk 1]
```diff
@@ -20,8 +20,7 @@ mod test_conf_change;
 mod test_lease_read;
 mod test_merge;
 mod test_multi;
-// TODO: test prevote.
-// mod test_prevote;
+mod test_prevote;
```
Comment (Hoverbear) [inline tests/raftstore_cases/mod.rs]: Oh dear. This was missed in https://github.com/pingcap/tikv/pull/3289 :(

**Comment by siddontang** [comment]
is the test reliable here now? @Hoverbear

**Comment by siddontang** [comment]
or do we still need to remove them?

**Comment by Hoverbear** [comment]
@siddontang https://github.com/pingcap/tikv/pull/3386 makes them reliable.

**Comment by hicqu** [comment]
ping @overvenus @Hoverbear PTAL thanks.

**Comment by siddontang** [comment]
Please approve it @Hoverbear

**Pull request merged**